src/index.js

```diff
@@ -3,12 +3,13 @@ import {queries, waitForElement} from 'dom-testing-library'
 const commands = Object.keys(queries).map(queryName => {
   return {
     name: queryName,
-    command: (cy, ...args) => {
+    command: (cy, text, options = {}) => {
+      const { timeout = 3000 } = options;
       const queryImpl = queries[queryName]
       const baseCommandImpl = doc =>
-        waitForElement(() => queryImpl(doc, ...args), {
+        waitForElement(() => queryImpl(doc, text, options), {
```

Comment:
This call isn't quite safe. What if `queryImpl` takes other set of arguments, not just `text` followed by `options`? The original idea was to propagate all the query options verbatim to `queryImpl`. I would consider `queryImpl.length` (the function arity) to see how many arguments it actually takes, and passed the respective slice of `args` to it; if there's one more argument remaining, that's our Cypress command `options` with `timeout` and potentially other stuff such as `log: false`.

Relying on function arity may be unreliable if the `queryImpl` uses variable number of `...args` (i.e. `arguments`) instead of defining all arguments explicitly.

Reply:
Good point. However, I prefer to have a single options object, so there's no need to keep in mind which options go into which object. Do you agree?

Reply:
I prefer the `...args` for stuff like this if we can get away with it.

Reply:
> I would consider queryImpl.length (the function arity) to see how many arguments it actually takes, and passed the respective slice of args to it; if there's one more argument remaining, that's our Cypress command options with timeout and potentially other stuff such as log: false.

So if I understand this correctly, the function arity of our command function is the function arity of the query function + 1, right?
If the query function takes two arguments; `text` and `options` and `options` is optional, you'd have to pass in `("text", undefined, { timeout: 10000 })` to set a timeout?

I prefer the API to have a single options object. For that to work and still use `...args`, we'd have to make the assumption that the `options` parameter is the last argument. In that case, I can retrieve it using `args[args.length -1]` and destructure it like already in the PR. So I agree that `...args` is better, but I think the command function shouldn't have an extra options object just for the Cypress related options. What are your thoughts on that?

Reply:
Honestly I don't care all that much. Let's just go with what you have here and we can change it in the future if it needs changing. I think I'm good with this PR as-is :)

Reply:
Okay, whatevs. I'm just saying that `queryImpl` is not guaranteed to be that generic to always take `(doc, text, options)`.

Reply:
Hmmm... I think that's right. In that case, below where we're using `text` and `options` in `message`, do we need to make that more generic somehow to support different APIs?

Reply:
> Hmmm... I think that's right. In that case, below where we're using `text` and `options` in `message`, do we need to make that more generic somehow to support different APIs?

Yes, that, too. There was a generic solution before: just pass all the `args` for `message` (Cypress can handle arrays in the `message` as far as I remember).